src/main/java/org/apache/maven/model/jdom/JDomDependency.java

```diff
@@ -25,52 +25,86 @@
 import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_OPTIONAL;
 import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_SCOPE;
 import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_SYSTEM_PATH;
 import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_TYPE;
+import static org.apache.maven.model.jdom.util.JDomCfg.POM_ELEMENT_VERSION;
 import static org.apache.maven.model.jdom.util.JDomUtils.getChildElementTextTrim;
 import static org.apache.maven.model.jdom.util.JDomUtils.rewriteElement;
+import static org.codehaus.plexus.util.StringUtils.defaultString;
+import static org.codehaus.plexus.util.StringUtils.trim;
 
 /**
- * JDom implementation of poms DEPENDENCY element
+ * JDOM implementation of the {@link Dependency} class. It holds the child elements of the Maven POMs {@code dependency}
+ * element.
  *
  * @author Robert Scholte (for <a href="https://github.com/apache/maven-release/">Maven Release projct</a>, version 3.0)
+ * @author Marc Rohlfs, CoreMedia AG
  */
-public class JDomDependency extends Dependency implements JDomBacked, MavenCoordinate {
+public class JDomDependency extends Dependency implements JDomBacked {
 
-  private final Element jdomElement;
+  private static final long serialVersionUID = -6973299112773078102L;
 
-  private final MavenCoordinate coordinate;
+  private final Element jdomElement;
 
-  public JDomDependency(Element jdomElement) {
+  JDomDependency(Element jdomElement) {
     this.jdomElement = jdomElement;
-    this.coordinate = new JDomMavenCoordinate(jdomElement);
-  }
 
-  @Override
-  public String getArtifactId() {
-    return coordinate.getArtifactId();
+    super.setArtifactId(getChildElementTextTrim(POM_ELEMENT_ARTIFACT_ID, jdomElement));
+    super.setClassifier(getChildElementTextTrim(POM_ELEMENT_CLASSIFIER, jdomElement));
+    super.setGroupId(getChildElementTextTrim(POM_ELEMENT_GROUP_ID, jdomElement));
+    super.setOptional(getChildElementTextTrim(POM_ELEMENT_OPTIONAL, jdomElement));
+    super.setScope(getChildElementTextTrim(POM_ELEMENT_SCOPE, jdomElement));
+    super.setSystemPath(getChildElementTextTrim(POM_ELEMENT_SYSTEM_PATH, jdomElement));
+    super.setVersion(getChildElementTextTrim(POM_ELEMENT_VERSION, jdomElement));
+
+    String type = getChildElementTextTrim(POM_ELEMENT_TYPE, jdomElement);
+    if (type != null) {
+      super.setType(type);
+    }
   }
 
-  @Override
-  public void setArtifactId(String artifactId) {
-    coordinate.setArtifactId(artifactId);
+  JDomDependency(Element jdomElement, Dependency dependency) {
+    this.jdomElement = jdomElement;
+
+    setArtifactId(dependency.getArtifactId());
+    setClassifier(dependency.getClassifier());
+    setGroupId(dependency.getGroupId());
+    setOptional(dependency.getOptional());
+    setScope(dependency.getScope());
+    setSystemPath(dependency.getSystemPath());
+    setVersion(dependency.getVersion());
+
+    String type = dependency.getType();
+    if (!"jar".equals(type)) {
```

Comment:
Isn't there a constant for `jar`?

Reply:
Didn't find one, but we may introduce one.